connectors/sources/mysql.py

```diff
@@ -159,6 +159,18 @@ async def with_connection_pool(self):
             connection_pool.close()
             await connection_pool.wait_closed()
 
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+    )
+    async def get_all_table_names(self):
+        async with self.with_connection_pool() as connection_pool:
+            async with connection_pool.acquire() as connection:
```

Comment:
I just find it a bit weird that we're unable to re-use connection pool between methods. I think it _might_ get problematic. Do you think we can make sure that the client has one connection pool throughout all calls?

Reply:
Yes, I planned to add it as a field and reuse it throughout the instance methods.